qcodes/tests/dataset/test_plotting.py

```diff
@@ -59,35 +60,25 @@ def test_rescaled_ticks_and_units(scale, unit,
     }
 
     ticks_formatter, label = _make_rescaled_ticks_and_units(data_dict)
-
-    expected_prefix = _ENGINEERING_PREFIXES[scale]
+    if unit in _UNITS_FOR_RESCALING:
+        expected_prefix = _ENGINEERING_PREFIXES[scale]
+    else:
+         if scale != 0:
+             expected_prefix = f'$10^{{{scale:.0f}}}$ '
+         else:
+             expected_prefix = ''
     if param_label == '':
-        assert f"{param_name} ({expected_prefix}{unit})" == label
+        base_label = param_name
     else:
-        assert f"{param_label} ({expected_prefix}{unit})" == label
+        base_label = param_label
+    postfix = expected_prefix + unit
+    if postfix != '':
+        assert f"{base_label} ({postfix})" == label
+    else:
+        assert f"{base_label}" == label
```

Comment:
the test method became too complex: it basically mirrors the logic of the function under test. Such a test becomes difficult to read and more difficult to modify. Let's be careful next time. As the initial author of the test, i should've also been more careful.

Reply:
Good point. Thanks for pointing that out, I got carried away ;-)